lib/discourse_code_review/state/commit_topics.rb

```diff
@@ -94,10 +94,16 @@ def ensure_commit(commit:, merged:, repo_name:, user:, category_id:, followees:)
 
             tags << SiteSetting.code_review_commit_tag
 
+            truncated_title =
+              if Topic.fancy_title(title).length > Topic.max_fancy_title_length
+                Topic.fancy_title(title).truncate(Topic.max_fancy_title_length)
+              else
+                title
```

Comment:
Wait, isn't `Topic.fancy_title` already checking the lenght?

Reply:
Not really from my understanding.

https://github.com/discourse/discourse/blob/master/app/models/topic.rb#L442

```ruby
  def self.fancy_title(title)
    return unless escaped = ERB::Util.html_escape(title)
    fancy_title = Emoji.unicode_unescape(HtmlPrettify.render(escaped))
    fancy_title.length > Topic.max_fancy_title_length ? escaped : fancy_title
  end
```

If it's over the length it will return escaped but not doing any truncation.

Reply:
Then, how about always truncating? This also avoid calling `Topic.fancy_title` twice.

```ruby
fancy_title = Topic.fancy_title(title).truncate(Topic.max_fancy_title_length)
```

Reply:
Because title != fancy_title

Reply:
Now that I think more about it maybe :

```
Topic.fancy_title(title).truncate(Topic.max_fancy_title_length)
```

would be enough because it will indeed actually mostly act as a `title.truncate` if under limit